client/state/data-layer/wpcom/comments/index.js

```diff
@@ -36,6 +63,38 @@ export const fetchPostComments = ( { dispatch, getState }, action ) => {
 	}, action ) );
 };
 
+export const writePostComment = ( { dispatch }, action ) => {
+	const { siteId, postId, parentCommentId, commentText } = action;
+	const placeholder = createPlaceholderComment( commentText, postId, parentCommentId );
+
+	// Insert a placeholder
+	dispatch( {
+		type: COMMENTS_RECEIVE,
+		siteId,
+		postId,
+		comments: [ placeholder ],
+		skipSort: !! parentCommentId
+	} );
+
+	const path = !! parentCommentId
+		? `/sites/${ siteId }/comments/${ parentCommentId }/replies/new`
+		: `/sites/${ siteId }/posts/${ postId }/replies/new`;
```

Comment:
this is tricky for our naming convention, but it would imply either way that this file should be differently named. I would probably recommend based on the name "resource" which I just pulled out of thin air…

```
state/data-layer/wpcom/sites/resources/replies/new/index.js
```

the naming convention is part of the system to make it easier to associate API calls with the code calling it. this is the first big challenge to that convention